From 2c34564969925c82f475de411e9416a7875dcf64 Mon Sep 17 00:00:00 2001
From: Gilles DOFFE <g.doffe@gmail.com>
Date: Sat, 16 Sep 2023 18:47:34 +0200
Subject: [PATCH] drivers/pcf857x: fix returns on void functions
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The pcf857x_gpio_write() is also returning void as pcf857x_gpio_*
functions.

This fixes following errors using native architecture:
RIOT/drivers/pcf857x/pcf857x.c:319:12: error: ISO C forbids ‘return’
with expression, in function returning void [-Werror=pedantic]
  319 |     return pcf857x_gpio_write(dev, pin, 0);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Gilles DOFFE <g.doffe@gmail.com>
---
 drivers/pcf857x/pcf857x.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/pcf857x/pcf857x.c b/drivers/pcf857x/pcf857x.c
index d74c4a8a18..e5b149d962 100644
--- a/drivers/pcf857x/pcf857x.c
+++ b/drivers/pcf857x/pcf857x.c
@@ -316,19 +316,19 @@ void pcf857x_gpio_write(pcf857x_t *dev, gpio_t pin, int value)
 void pcf857x_gpio_clear(pcf857x_t *dev, gpio_t pin)
 {
     DEBUG_DEV("pin=%u", dev, pin);
-    return pcf857x_gpio_write(dev, pin, 0);
+    pcf857x_gpio_write(dev, pin, 0);
 }
 
 void pcf857x_gpio_set(pcf857x_t *dev, gpio_t pin)
 {
     DEBUG_DEV("pin=%u", dev, pin);
-    return pcf857x_gpio_write(dev, pin, 1);
+    pcf857x_gpio_write(dev, pin, 1);
 }
 
 void pcf857x_gpio_toggle(pcf857x_t *dev, gpio_t pin)
 {
     DEBUG_DEV("pin=%u", dev, pin);
-    return pcf857x_gpio_write(dev, pin, (dev->out & (1 << pin)) ? 0 : 1);
+    pcf857x_gpio_write(dev, pin, (dev->out & (1 << pin)) ? 0 : 1);
 }
 
 /** Functions for internal use only */
-- 
2.37.0

